src/containers/StructurePage/components/FolderItem.jsx

```diff
@@ -7,27 +7,62 @@
  */
 
 import React from 'react';
-import { string, bool } from 'prop-types';
+import { string, bool, arrayOf, object, shape } from 'prop-types';
 import { Folder } from 'ndla-icons/editor';
 import { Link } from 'react-router-dom';
 import BEMHelper from 'react-bem-helper';
+import SettingsMenu from './SettingsMenu';
 
 const classes = new BEMHelper({
   name: 'folder',
   prefix: 'c-',
 });
 
-const FolderItem = ({ title, path, active }) => (
-  <Link to={path} {...classes('')}>
-    <Folder color={'#70A5DA'} />
-    <span {...classes('title', active && 'active')}>{title}</span>
-  </Link>
-);
+class FolderItem extends React.PureComponent {
+  render() {
+    const { name, path, active, topics = [], params, id, ...rest } = this.props;
```

Comment:
`...rest` makes it hard to understand/read which props flows through FolderItem. They are also not listed in propTypes.

Reply:
That was kind of the point, as the list of props will grow very large with more and more settings. The propTypes are declared where they are used, I don't see the need to declare them multiple times? It will be slightly harder to refactor if SettingsMenu needs to use different props in the future, but you gain readability and prevents props from being "stopped" along the way down :)